metadata-ingestion/src/datahub/ingestion/source/nifi.py

```diff
@@ -694,8 +781,11 @@ def fetch_provenance_events(
 
         if provenance_response.ok:
             provenance = provenance_response.json().get("provenance", {})
+            total = provenance.get("results", {}).get("total")
```

Comment:
Would these counts be available in provenance results before verifying that fetching provenance results has finished - which is done below in while loop ?

Reply:
That's a great catch - while it doesn't cause any exception it actually always returns `0` in such case for `total` and `totalCount`. Moved the retrieval of these values to back where they were.